mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,491 @@
+                                      "work in progress.")
+            #warning("results_backend is set to 'native'. This has been known"
+            #        "to result in longer run time when working with big datasets.")
+        if self.params.results_backend == 'hdf5' and \
+                not externals.exists('h5py'):
+            raise RuntimeError("The 'hdf5' module is required for "
+                               "when results_backend is set to 'hdf5'")
+
+    def _proc_block(self, block, datasets, measure, qe, seed=None, iblock='main'):
+        if seed is not None:
+            mvpa2.seed(seed)
+        if __debug__:
+            debug('SLC', 'Starting computing block for %i elements' % len(block))
+        bar = ProgressBar()
+        projections = [csc_matrix((self.nfeatures, self.nfeatures),
+                                  dtype=self.params.dtype) for isub in range(self.ndatasets)]
+        for i, node_id in enumerate(block):
+            # retrieve the feature ids of all features in the ROI from the query
+            # engine
+
+            # Find the neighborhood for that selected nearest node
+            roi_feature_ids = qe[node_id]
+            # if qe returns zero-sized ROI for any subject, pass...
+            if len(roi_feature_ids) == 0:
+                continue
+            # selecting neighborhood for all subject for hyperalignment
+            ds_temp = [sd[:, roi_feature_ids] for sd in datasets]
+            roi_seed = np.array(roi_feature_ids) == node_id
+            ds_temp[self.params.ref_ds].fa['roi_seed'] = roi_seed
+            if __debug__:
+                msg = 'ROI (%i/%i), %i features' % (i + 1, len(block), len(roi_seed))
+                debug('SLC', bar(float(i + 1) / len(block), msg), cr=True)
+            hmappers = measure(ds_temp)
+            hmappers = hmappers.samples
+            for isub in range(len(hmappers)):
+                if not self.params.combine_neighbormappers:
+                    I = roi_feature_ids
+                    #J = [roi_feature_ids[node_id]] * len(roi_feature_ids)
+                    J = [node_id] * len(roi_feature_ids)
+                    V = hmappers[isub][0]['proj'].tolist()
+                else:
+                    I = []
+                    J = []
+                    V = []
+                    for f2 in xrange(len(roi_feature_ids)):
+                        I += roi_feature_ids
+                        J += [roi_feature_ids[f2]] * len(roi_feature_ids)
+                        V += hmappers[isub][0]['proj'][:, f2].tolist()
+                proj = coo_matrix((V, (I, J)),
+                                  shape=(max(self.nfeatures, max(I)+1), max(self.nfeatures, max(J)+1)),
+                                  dtype=self.params.dtype)
+                proj = proj.tocsc()
+                # Cleaning up the current subject's projections to free up memory
+                hmappers[isub, ] = [[]]*hmappers.shape[1]
```

Comment:
A PR with slight tune ups (without changing functionality) is coming but this line is a bit troublesome... may be irrelevant since may be later each entry overriden in full, but what happens is that you are generating a list of the same (not copies but literally the same) empty lists, so modifying one would change the others:
```ipython
In [1]: l = [[]] * 2

In [2]: l
Out[2]: [[], []]

In [3]: l[0].append(2)

In [4]: l
Out[4]: [[2], [2]]

# the same happens if within array
In [6]: a = np.empty(shape=(2,), dtype=object)

In [7]: a[0] = [[]] * 2

In [8]: a
Out[8]: array([[[], []], None], dtype=object)

In [9]: a[0][0].append(2)

In [10]: a
Out[10]: array([[[2], [2]], None], dtype=object)

```
once again -- may be all irrelevant but I will fix it up to be a proper `[[] for _ in xrange(hmappers.shape[1]]`